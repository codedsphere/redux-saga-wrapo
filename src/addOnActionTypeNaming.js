import { ReducerDefaultActionTypes } from './constants/reducer';

export function addOnActionTypeNaming(
  ActionTypes,
  isReducer = false
) {
  for (const [actionTypesKey, actionTypesValue] of Object.entries(
    ActionTypes,
  )) {
    if (isReducer) {
      for (const [
        defaultActionTypesKey,
        defaultActionTypesValue,
      ] of Object.entries(ReducerDefaultActionTypes)) {
        ActionTypes[actionTypesKey][
          defaultActionTypesKey
        ] = defaultActionTypesValue;
      }
      ActionTypes[actionTypesKey].namespace = actionTypesKey;
    }

    for (const [innerActionTypesKey, innerActionTypesValue] of Object.entries(
      actionTypesValue,
    )) {
      if (innerActionTypesKey !== 'namespace') {
        ActionTypes[actionTypesKey][
          innerActionTypesKey
        ] = `${actionTypesKey}_${innerActionTypesValue}`;
      }
    }
  }
  if (isReducer) {
    if (global.ActionType) {
      global.ActionType = { ...ActionTypes, ...global.ActionType };
    }
    else {
      global.ActionType = ActionTypes;
    }

    return global.ActionType;
  }
  return ActionTypes;
}
