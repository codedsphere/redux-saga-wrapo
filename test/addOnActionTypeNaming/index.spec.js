import rootReducer from '../__mocks__/rootReducer';
import rootSaga from '../__mocks__/rootSaga';

import { addOnActionTypeNaming } from 'addOnActionTypeNaming';

describe('addOnActionTypeNaming', () => {
  it('should have a addOnActionTypeNaming', () => {
    
    const ReducerActionTypes = addOnActionTypeNaming(
      rootReducer,
      true,
    );
    const SagaActionTypes = addOnActionTypeNaming(
      rootSaga,
      false,
    );

    expect(ReducerActionTypes).toMatchSnapshot();
    expect(SagaActionTypes).toMatchSnapshot();
  });
});

