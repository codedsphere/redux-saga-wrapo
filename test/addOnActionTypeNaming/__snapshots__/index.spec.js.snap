// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addOnActionTypeNaming should have a addOnActionTypeNaming 1`] = `
Object {
  "appModel": Object {
    "INITIALIZE_ALL": "appModel_INITIALIZE_ALL",
    "INITIALIZE_STATE": "appModel_INITIALIZE_STATE",
    "REMOVE_STATE": "appModel_REMOVE_STATE",
    "UPDATE_STATE": "appModel_UPDATE_STATE",
    "namespace": "appModel",
  },
  "exampleModel": Object {
    "INITIALIZE_ALL": "exampleModel_INITIALIZE_ALL",
    "INITIALIZE_STATE": "exampleModel_INITIALIZE_STATE",
    "Other_Reducer": "exampleModel_Other_Reducer",
    "REMOVE_STATE": "exampleModel_REMOVE_STATE",
    "UPDATE_STATE": "exampleModel_UPDATE_STATE",
    "namespace": "exampleModel",
  },
  "formModel": Object {
    "INITIALIZE_ALL": "formModel_INITIALIZE_ALL",
    "INITIALIZE_STATE": "formModel_INITIALIZE_STATE",
    "REMOVE_STATE": "formModel_REMOVE_STATE",
    "UPDATE_STATE": "formModel_UPDATE_STATE",
    "namespace": "formModel",
  },
  "themeModel": Object {
    "INITIALIZE_ALL": "themeModel_INITIALIZE_ALL",
    "INITIALIZE_STATE": "themeModel_INITIALIZE_STATE",
    "Other_Reducer": "themeModel_Other_Reducer",
    "REMOVE_STATE": "themeModel_REMOVE_STATE",
    "UPDATE_STATE": "themeModel_UPDATE_STATE",
    "namespace": "themeModel",
  },
  "userModel": Object {
    "INITIALIZE_ALL": "userModel_INITIALIZE_ALL",
    "INITIALIZE_STATE": "userModel_INITIALIZE_STATE",
    "Other_Reducer": "userModel_Other_Reducer",
    "REMOVE_STATE": "userModel_REMOVE_STATE",
    "UPDATE_STATE": "userModel_UPDATE_STATE",
    "namespace": "userModel",
  },
}
`;

exports[`addOnActionTypeNaming should have a addOnActionTypeNaming 2`] = `[Function]`;
